trilby-cli/src/Trilby/Update.hs

```diff
@@ -3,16 +3,76 @@
 module Trilby.Update (update) where
 
 import Trilby.Update.Options
+import Turtle (readlink)
+import Data.Text qualified as Text
 import Prelude
+import Data.List.Extra (split)
 
 update :: UpdateOpts Maybe -> App ()
-update (askOpts -> opts) = inDir "/etc/trilby" do
+update (askOpts -> opts) = do
     whenM opts.flakeUpdate $ rawCmd_ ["nix", "flake", "update", "--accept-flake-config"]
-    withTrace rawCmd_ ["nixos-rebuild", "build", "--flake", ".", "--accept-flake-config"]
-    rawCmd_ ["nvd", "diff", "/run/current-system", "result"]
+    hosts <- fromListSafe Localhost <$> (fromText <$$> opts.hosts)
+    for_ @NonEmpty hosts $ updateHost opts
+
+updateHost :: UpdateOpts App -> Host -> App ()
+updateHost opts host = inDir "/etc/trilby" do
+    perform NixosRebuildBuild
+    resultPath <- readlink "./result"
```

Comment:
Sadly this doesn’t work. It would appear that `nixos-rebuild` does not produce a `./result` symlink when given a `--target-host`.

We have a few bad options:
- build once without target-host, and then again with it (slow double eval)
- build without target-host and then use `nix copy` manually (annoying)
- build with `nix build` directly
- patch nixos-rebuild upstream
- somehow query the local nix store, or eke out the path from the output (hacky)